app/src/main/java/woowacourse/shopping/data/repository/CouponRepositoryImpl.kt

```
@@ -0,0 +1,16 @@
+
+import woowacourse.shopping.data.datasource.CouponDataSource
+import woowacourse.shopping.data.dto.order.toDomain
+import woowacourse.shopping.order.Coupon
+
+class CouponRepositoryImpl(
+    private val couponDataSource: CouponDataSource,
+) : CouponRepository {
+    override suspend fun getCoupons(): List<Coupon> =
+        try {
+            couponDataSource.fetchCoupons().map { it.toDomain() }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)